<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>handleCoroutineException - kotlinx-coroutines-core</title>
  <meta name="description" content="Library support for kotlin coroutines">
  <link rel="stylesheet" href="/kotlinx.coroutines/assets/main.css">
  <link rel="canonical" href="https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/handle-coroutine-exception.html">
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="/kotlinx.coroutines/assets/js/api.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-47631155-3', 'auto');
    ga('send', 'pageview');
  </script>
</head>

  <body>
    <header class="site-header" role="banner">
  <div class="wrapper">
    <a class="site-title" href="/kotlinx.coroutines/">kotlinx.coroutines</a>
  </div>
</header>

    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <div class="api-docs-breadcrumbs"><a href="../index.html">kotlinx-coroutines-core</a> / <a href="index.html">kotlinx.coroutines</a> / <a href="./handle-coroutine-exception.html">handleCoroutineException</a></div>

<h1 id="handlecoroutineexception">handleCoroutineException</h1>

<div class="signature"><code><span class="keyword">fun </span><span class="identifier">handleCoroutineException</span><span class="symbol">(</span><br />&nbsp;&nbsp;&nbsp;&nbsp;<span class="parameterName" id="kotlinx.coroutines$handleCoroutineException(kotlin.coroutines.CoroutineContext, kotlin.Throwable)/context">context</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/-coroutine-context/index.html"><span class="identifier">CoroutineContext</span></a><span class="symbol">, </span><br />&nbsp;&nbsp;&nbsp;&nbsp;<span class="parameterName" id="kotlinx.coroutines$handleCoroutineException(kotlin.coroutines.CoroutineContext, kotlin.Throwable)/exception">exception</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><br /><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code> <a href="https://github.com/kotlin/kotlinx.coroutines/tree/master/kotlinx-coroutines-core/common/src/CoroutineExceptionHandler.kt#L21">(source)</a></div>

<p>Helper function for coroutine builder implementations to handle uncaught and unexpected exceptions in coroutines,
that could not be otherwise handled in a normal way through structured concurrency, saving them to a future, and
cannot be rethrown. This is a last resort handler to prevent lost exceptions.</p>

<p>If there is <a href="-coroutine-exception-handler/index.html">CoroutineExceptionHandler</a> in the context, then it is used. If it throws an exception during handling
or is absent, all instances of <a href="-coroutine-exception-handler/index.html">CoroutineExceptionHandler</a> found via <a href="#">ServiceLoader</a> and
<a href="http://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#uncaughtExceptionHandler">Thread.uncaughtExceptionHandler</a> are invoked.</p>


      </div>
    </main>
    <footer class="site-footer">
  <!-- empty -->
</footer>

  </body>
</html>

